Add a NOT_CONTROLABLE example (#89)
flexiblepower/s2-documentation: website/docs/examples/nocontrol.md

---
title: "Example: Non controllable assets"
hide_title: true
---

> This page serves as a guide for implementing an S2 RM for a non-flexible device, or to give a CEM developer a better understanding of what to expect when controlling the energy flexibility of a non-flexible device. It provides some example S2 messages for same (fictional) non-flexible devices.

Reviewer: ```suggestion
This page serves as a guide for implementing an S2 RM for a non-flexible device, or to give a CEM developer a better understanding of what to expect when taking into account information from a non-flexible device. It provides some example S2 messages for same (fictional) non-flexible devices.
```

> This page serves as a guide for implementing an S2 RM for a non-flexible device, or to give a CEM developer a better understanding of what to expect when controlling the energy flexibility of a non-flexible device. It provides some example S2 messages for same (fictional) non-flexible devices.

Reviewer: ```suggestion
This page serves as a guide for implementing an S2 RM for a non-flexible device, or to give a CEM developer a better understanding of what to expect when controlling the energy flexibility of a non-flexible device. It provides some example S2 messages for same (fictional) non-flexible devices. Some examples of non-flexible devices are a TV, an electric cooking appliance, lights and a non-curtailable PV system.
```

```json
{
  "message_type": "PowerMeasurement",
  "message_id": "6652c78e-57c8-4e04-807f-685b6b3fde17",
  "measurement_timestamp": "2019-08-24T14:15:22Z",
  "values": [
    {
      "commodity_quantity": "ELECTRIC.POWER.L1",
      "value": 3450.6
    }
  ]
}
```

### RM -> CEM: PowerForecast
It can be quite useful for a CEM to know in advance what the future energy usage of a device will be. This is true for both controllable and non-controllable devices. Such a forecast can be used by an energy management algorithm to establish the base load it will have to work with. The data for this power forecast could for example be retrieved from a weather service that can calculate this forecast based the location and orientation of the asset. A PowerForcast can also incorporate uncertainty of the forecast in the values.

```json
{
  "message_type": "PowerForecast",
  "message_id": "bde674ac-66ef-4fed-8b12-d01b8404e10d",
  "start_time": "2024-08-24T14:00:00Z",
  "elements": [
    {
      "duration": 3600000,
      "power_values": [
        {
          "value_upper_limit": -3500.0,
          "value_upper_95PPR": -3460.0,
          "value_upper_68PPR": -3455.0,

>           "value_expected": -3450.1,

Reviewer: It's a bit strange that the measurement is a positive value and the forecast negative. I suggest to remove the minus signs.
